alts/src/main/java/io/grpc/alts/HandshakerServiceChannel.java

```diff
@@ -16,57 +16,51 @@
+ * Class for creating a single shared gRPC channel to the ALTS Handshaker Service using
+ * SharedResourceHolder. The channel to the handshaker service is local and is over plaintext. Each
+ * application will have at most one connection to the handshaker service.
  */
 final class HandshakerServiceChannel {
-  // Default handshaker service address.
-  private static String handshakerAddress = "metadata.google.internal:8080";
-  // Shared channel to ALTS handshaker service.
-  private static ManagedChannel channel = null;
 
-  // Construct me not!
-  private HandshakerServiceChannel() {}
+  static final HandshakerChannelResource SHARED_HANDSHAKER_CHANNEL =
+      new HandshakerChannelResource();
+
+  static class HandshakerChannelResource implements Resource<ManagedChannel> {
 
-  // Sets handshaker service address for testing and creates the channel to the handshaker service.
-  public static synchronized void setHandshakerAddressForTesting(String handshakerAddress) {
-    Preconditions.checkState(
-        channel == null || HandshakerServiceChannel.handshakerAddress.equals(handshakerAddress),
-        "HandshakerServiceChannel already created with a different handshakerAddress");
-    HandshakerServiceChannel.handshakerAddress = handshakerAddress;
-    if (channel == null) {
-      channel = createChannel();
+    private String handshakerAddress = "metadata.google.internal:8080";
+    private static final String NAME = "grpc-alts-handshaker-service-channel";
+
+    public void setHandshakerAddressForTesting(String handshakerAddress) {
+      this.handshakerAddress = handshakerAddress;
     }
-  }
 
-  /** Create a new channel to ALTS handshaker service, if it has not been created yet. */
-  private static ManagedChannel createChannel() {
-    /* Use its own event loop thread pool to avoid blocking. */
-    ThreadFactory clientThreadFactory = new DefaultThreadFactory("handshaker pool", true);
-    ManagedChannel channel =
-        NettyChannelBuilder.forTarget(handshakerAddress)
-            .directExecutor()
-            .eventLoopGroup(new NioEventLoopGroup(1, clientThreadFactory))
-            .usePlaintext()
-            .build();
-    return channel;
-  }
+    @Override
+    public ManagedChannel create() {
+      /* Use its own event loop thread pool to avoid blocking. */
+      ThreadFactory clientThreadFactory = new DefaultThreadFactory("handshaker pool", true);
+      return NettyChannelBuilder.forTarget(handshakerAddress)
+          .directExecutor()
+          .eventLoopGroup(new NioEventLoopGroup(1, clientThreadFactory))
+          .usePlaintext()
+          .build();
+    }
+
+    @Override
+    public void close(ManagedChannel instance) {
+      instance.shutdown();
+    }
 
-  public static synchronized ManagedChannel get() {
-    if (channel == null) {
-      channel = createChannel();
+    @Override
+    public String toString() {
+      return NAME;
```

Comment:
Feel free to have the literal inline. `GrpcUtil.SHARED_CHANNEL_EXECUTOR` uses a constant `NAME` because it uses the string twice in the class.